tests/test_workers/test_pkg_managers/test_pip.py

```diff
@@ -654,3 +654,466 @@ def test_get_version_attr_with_package_dir(
+                    """\
+                    foo = "bar"
+
+                    setup({what}=foo)
+                    """
+                ),
+                "bar",
+                ["Pseudo-path: Module.body[1] -> Expr(#3).value", "Found variable 'foo': 'bar'"],
+            ),
+            (
+                dedent(
+                    """\
+                    foo = "bar"
+
+                    if True:
+                        setup({what}=foo)
+                    """
+                ),
+                "bar",
+                [
+                    "Pseudo-path: Module.body[1] -> If(#3).body[0] -> Expr(#4).value",
+                    "Found variable 'foo': 'bar'",
+                ],
+            ),
+            (
+                # Variable will be found only if it is in the same branch
+                dedent(
+                    """\
+                    if True:
+                        foo = "bar"
+                    else:
+                        setup({what}=foo)
+                    """
+                ),
+                None,
+                [
+                    "Pseudo-path: Module.body[0] -> If(#1).orelse[0] -> Expr(#4).value",
+                    "Variable 'foo' not found along the setup call branch",
+                ],
+            ),
+            (
+                dedent(
+                    """\
+                    if True:
+                        foo = "bar"
+                        setup({what}=foo)
+                    """
+                ),
+                "bar",
+                [
+                    "Pseudo-path: Module.body[0] -> If(#1).body[1] -> Expr(#3).value",
+                    "Found variable 'foo': 'bar'",
+                ],
+            ),
+            (
+                # Try statements are kinda special, because not only do they have 3 bodies,
+                # they also have a list of 'handlers' (1 for each except clause)
+                dedent(
+                    """\
+                    try:
+                        pass
+                    except A:
+                        foo = "bar"
+                    except B:
+                        setup({what}=foo)
+                    else:
+                        pass
+                    finally:
+                        pass
+                    """
+                ),
+                None,
+                [
+                    (
+                        "Pseudo-path: Module.body[0] -> Try(#1).handlers[1] "
+                        "-> ExceptHandler(#5).body[0] -> Expr(#6).value"
+                    ),
+                    "Variable 'foo' not found along the setup call branch",
+                ],
+            ),
+            (
+                dedent(
+                    """\
+                    try:
+                        pass
+                    except A:
+                        pass
+                    except B:
+                        foo = "bar"
+                        setup({what}=foo)
```

Comment:
interesting. What happens when another branch has a similar block (let's say this is replicated in except A with another value attributed to foo)? We will just get the last occurrence, is that it?

This sounds good for now (hoping there is no sane use  cases for this), we just need to make that clear somewhere.

Reply:
Do you mean something like this:
```python
try:
    pass
except A:
    foo = "1.0.0"
    setup(version=foo)
except B:
    foo = "2.0.0"
    setup(version=foo)
```

In this case we will find the first setup call (in `except A`) and use the value of foo defined in the same block (or any parent block). I'm hoping there is no legitimate use case for calling setup() in 2 or more places in the file, we would have no way to determine which one gets called anyway.

So it basically comes down to: We expect the file to have exactly one setup() call, if there are 2 or more we will only process the first one

I'll add this to the class docstring if this is what you meant

Reply:
Added to the docstring of `_setup_branch` property